internal/plugin/plugin_manager.go

```diff
@@ -15,22 +15,15 @@ import (
 func LoadPlugins(agentConfig *config.Config, slogger *slog.Logger) []bus.Plugin {
 	plugins := []bus.Plugin{}
 
-	processMonitor := NewProcessMonitor(&ProcessMonitorParameters{
-		MonitoringFrequency: agentConfig.ProcessMonitor.MonitoringFrequency,
-	})
-
-	instanceMonitor := NewInstance(&InstanceParameters{})
+	processMonitor := NewProcessMonitor(agentConfig)
+	instanceMonitor := NewInstance()
 
 	configPlugin := NewConfig()
 
 	plugins = append(plugins, processMonitor, instanceMonitor, configPlugin)
 
 	if agentConfig.DataplaneAPI.Host != "" && agentConfig.DataplaneAPI.Port != 0 {
-		dataplaneServer := NewDataplaneServer(&DataplaneServerParameters{
-			Host:   agentConfig.DataplaneAPI.Host,
-			Port:   agentConfig.DataplaneAPI.Port,
-			Logger: slogger,
-		})
+		dataplaneServer := NewDataplaneServer(agentConfig, slogger)
```

Comment:
What's the difference between passing `slogger` around as a parameter and calling e.g. `slog.Info()` like we've done elsewhere?

Reply:
I only need the logger here for this plugin as I need it to configure the http server with the same logger created in main.go. See line 83.
`server.Use(sloggin.NewWithConfig(dps.logger, sloggin.Config{DefaultLevel: slog.LevelDebug}))`